tests/spread/extensions/env-injector/task.yaml

```diff
@@ -0,0 +1,84 @@
+summary: Build and run a basic hello-world snap using extensions
+
+systems:
+  - ubuntu-24.04*
+
+environment:
+
+  SNAPCRAFT_ENABLE_EXPERIMENTAL_EXTENSIONS: "1"
+  SNAP_DIR: ../snaps/env-injector-hello
+  SNAP: env-injector-hello
+
+prepare: |
+
+  #shellcheck source=tests/spread/tools/snapcraft-yaml.sh
+  . "$TOOLS_DIR/snapcraft-yaml.sh"
+  set_base "$SNAP_DIR/snap/snapcraft.yaml"
+
+restore: |
+
+  cd "$SNAP_DIR"
+  snapcraft clean
+  rm -f ./*.snap
+  rm -rf ./squashfs-root
+  #shellcheck source=tests/spread/tools/snapcraft-yaml.sh
+  . "$TOOLS_DIR/snapcraft-yaml.sh"
+  restore_yaml "snap/snapcraft.yaml"
+
+execute: |
+
+  cd "$SNAP_DIR"
+  snapcraft
+
+  unsquashfs "${SNAP}"_1.0_*.snap
+
+  # Check that the env-exporter program is present
+  [ -f squashfs-root/bin/command-chain/env-exporter ]
+  
+  # Check that the hello command is present
+  [ -f squashfs-root/usr/local/bin/hello ]
+
+  snap install "${SNAP}"_1.0_*.snap --dangerous
+
+  echo "[env-injector] Creating global envfile"
+  # Create global envfile
+  echo 'HELLO_WORLD=Hello World' >> envfile.env
+
+  echo "[env-injector] Creating app-specific envfile"
+  # Create app-specific envfile
+  echo 'SCOPED=Scoped' >> appenv.env
+
+  echo "[env-injector] Setting global env variable"
+  # Set env vars: Global
+  snap set env-injector-hello env.global="World"
+
+  echo "[env-injector] Setting app-specific env variable"
+  # Set env vars: specific to each app
+  snap set env-injector-hello apps.hello1.env.hello="Hello"
+  snap set env-injector-hello apps.hello2.env.specific="City"
+
+  echo "[env-injector] Rejecting invalid keys"
+  # Set env vars: specific to each app
+  # To check if key with dot will be rejected 
+  snap set env-injector-hello env.word.dot="wrong"
+
+  echo "[env-injector] Testing order of env vars"
+  # To test order of env vars
+  echo 'ORDER="From envfile"' >> envfile.env
+  snap set env-injector-hello apps.hello1.env.order="from app-specific"
+
+  echo "[env-injector] Load envfiles"
+  # Load global and app-specific envfiles
+  snap set env-injector-hello envfile="${SNAP_DIR}"/envfile.env
+  snap set env-injector-hello apps.hello1.envfile="${SNAP_DIR}"/appenv.env
+
+  echo "[env-injector] Run hello1 and hello2 apps"
+  # Run the hello command
+  # These will fail if the env vars are not set as expected
+  env-injector-hello.hello1
+  env-injector-hello.hello2
```

Comment:
I think the test evaluation shouldn't be inside the test applications in this case. They could just relay back what they see, and we can verify the correctness in this script, otherwise you end up with the test split in two parts, one in shell script and other in C making it harder to follow and modify.

Reply:
I agree! I'll change this to use `/usr/bin/env` inside the snap instead of the C app.

Reply:
Adressed on https://github.com/canonical/snapcraft/pull/4925/commits/9d95567ca5fe108e973551b841422d3dbc076ef8

Thank you for the suggestion!